sites/cheerpj/src/content/docs/11-guides/basic-server-setup.md

```diff
@@ -0,0 +1,186 @@
+
+However, with certain browsers you may encounter issues with CheerpJ when hosting the compressed library files. In rare cases an error can happen since we get a 206 in our network response, indicading partial content, but also with a Content-Encoding header.
+
+You will encounter the following console error in that case:
+
+> [!warning] Network error for http://localhost:8085/example/native.js: HTTP server returned compressed partial data. That should not happen. CheerpJ cannot run.
+
+To resolve this issue, you will need to disable compression for range requests or compression altogether. To disable compression in `npx serve`, use `npx serve -u`; however, this will only serve the original uncompressed .js files.
+
+For properly serving compressed files without the Content-Encoding header, consider utilizing NGINX, which will be discussed further in the next section.
+
+## NGINX Guide
+
+We won't go into detail on how to install and setup NGINX, but will focus on the configuration necessary to integrate NGINX seamlessly with CheerpJ.
+
+For instructions on installing and setting up NGINX, you can refer to their official [Beginners Guide](https://nginx.org/en/docs/beginners_guide.html).
+Please note that the installation process may vary depending on your operating system. For guidance on installing NGINX on Windows, visit this [documentation](https://nginx.org/en/docs/windows.html) and for other operating systems, refer to [this](https://nginx.org/en/docs/install.html).
+
+### Nginx configuration file
+
+This NGINX configuration template will work with most Java applications and CheerpJ for testing purposes.
+
+```nginx
+worker_processes  1;
+error_log   nginx_main_error.log info;
+pid nginx_user.pid;
+daemon off;
+
+events {
+	worker_connections  1024;
+}
+
+http {
+	access_log  nginx_access.log;
+	error_log   nginx_error.log info;
+
+	include       /etc/nginx/mime.types;
+	default_type  application/octet-stream;
+
+	sendfile        on;
+
+	server {
+		listen       8080;
+		server_name  localhost;
+
+		gzip on;
+		gzip_types      application/javascript application/wasm text/plain;
+
+		charset utf-8;
+
+		location / {
+			# this should point to your root application folder
+			root /home/path/to/java/app;
+
+			autoindex on;
+			absolute_redirect off;
+			index  index.html index.htm;
+			add_header 'Cross-Origin-Opener-Policy' 'same-origin' always;
+			add_header 'Cross-Origin-Embedder-Policy' 'require-corp' always;
+			add_header 'Cross-Origin-Embedder-Policy' 'credentialless' always;
+		}
+	}
+}
+```
+
+Let's take a closer look at some important settings required for CheerpJ and to also ensure proper file serving.
+
+By including NGINX's `mime.types` file, NGINX effectively identifies the file type for each response and sets the corresponding `Content-Type` header in HTTP responses. This enables browsers and clients to interpret content accurately,
+
+```nginx
+include       /etc/nginx/mime.types;
```

Comment:
This is not always at the same location and the configuration varies by system. We should instead use a fixed set of mimes of interest, we only actually care about wasm and js for compression